components/date-picker/PickerMixin.jsx

```diff
@@ -1,14 +1,23 @@
+import React from 'react';
 import objectAssign from 'object-assign';
 import defaultLocale from './locale/zh_CN';
 import DateTimeFormat from 'gregorian-calendar-format';
 import GregorianCalendar from 'gregorian-calendar';
 
 export default {
+  contextTypes: {
+    locale: React.PropTypes.object,
```

Comment:
please update demo for react-intl too:
http://yiminghe.me/learning-react/examples/react-intl.html?locale=en-US